botorch/optim/optimize_homotopy.py

```diff
@@ -3,14 +3,23 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
-from collections.abc import Callable
+from __future__ import annotations
+
+from typing import Any
 
 import torch
 from botorch.acquisition import AcquisitionFunction
 from botorch.optim.homotopy import Homotopy
 from botorch.optim.optimize import optimize_acqf
 from torch import Tensor
 
+from collections.abc import Callable
+
+from botorch.generation.gen import TGenCandidates
+from botorch.optim.initializers import (
+    TGenInitialConditions,
+)
```

Comment:
Could you run `ufmt format .` on the PR? This will fail the linter.

Reply:
Sure, Can I add a pre-commit yaml to have this be an auto hook?